### PR TITLE
More channels to instruments

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
and make most instruments with multiple channels have a name for the 1st channel rather than fallback to 'normal' (all but harp and the e-basses), having 'open' for brass, 'arco' for strings, 'pizzicato' for acoustic bass. Add 'mute' (and 'open') channel to all brass using trumpet sound, add 'pop' and 'slap' to bass guitar too.
See also https://musescore.org/en/node/269784

The addition to text palette may need some tweaks to apply to 2.3 or 2.2.1 too.

While at it, I'd like to add (more) channels for guitars, like 'muted', 'jazz', 'distortion', 'overdriven' and maybe 'harmonics', (the latter should be correct, but doesn't sound like harmonics to me) using the corresponding GM instruments

Maybe even add Soprano/Alto and Tenor/Bass channels to the Women and Men 'instruments' (and the closed score SATB templates), to ease muting/soloing those in Mixer

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
The addition to text palette may need some tweaks to apply to 2.2 too.
To make sure no string has been forgotten to put into instrumentsxml.h. ...share/instruments/generateTs.py should be run.

**Comment by IsaacWeiss** [comment]
I'd wait to add those guitar channels until the Mixer is redesigned to allow the collapsing of extraneous channels to show only one row for each instrument. As is, the Mixer already becomes very unwieldy with e.g. the string orchestra (five instruments, fifteen channels to scroll through), and so many channels for each guitar would be more trouble than its worth.

**Comment by Jojo-Schmitz** [comment]
Well, you'd rarely have multiple guitars in one score, do you? So 2 channels for an acoustic guitar (3 for steel strung), 5-6 for el. guitars shouldn't do much harm?

**Comment by chenlung** [comment]
Except my scores, @Jojo-Schmitz ;)

**Comment by Jojo-Schmitz** [comment]
@chenlung: with multiple I mean many, as many as there are string instruments in a string ensemble or even a full orchestra, not just 2 or 3 (and nothing else)

**Comment by MarcSabatella** [comment]
A few comments:

- I noticed you didn't touch the code where we special case "normal".  Have you tested much to be sure there are no unintended side effects here?  I saw quite a few places we explicitly key off "normal" as a channel name, we need to make sure nothing breaks as a result of this change.  I *think* the places I saw (just search the code for "normal", with quotes) were more about setting defaults than actually *requiring* it.  But it is important to be sure.

- Having arco and pizz. in the text palette already set up is great.  Tremolo is more a don't care; it's seldom notated using text.  As for the acoustic bass instrument, it's not going to be super common for people using this instrument (jazz mostly) to need these texts at all, but I'm good with the texts being arco & pizz., and people can edit them if they prefer something they think would be "jazzier".

- I for one am glad to see guitar channels added.  I agree that ideally there would be a way to collapse the mixer entries, but as long as we don't go *too* crazy, I think it's a welcome change.  I think far more people would welcome the ability to change sounds than would complain about a couple of extra mixer entries.  I suppose different guitar instruments might in theory benefit from different channels ("distorted" on the "Classical guitar" makes little sense :-).

**Comment by Jojo-Schmitz** [comment]
Haven't yet done anything for 'normal', ideas as to where and how are welcome ;-)
I'll do a guitar channels commit later, maybe tomorrow or the day after, running out of time currently

**Comment by IsaacWeiss** [comment]
Even for a relatively limited number of guitars in a larger score, the guitars would rapidly dwarf all the rest of the instruments put together. Two electric guitars in a ten-part ensemble...

Let people use instrument changes to add these channels when they need them.

**Comment by chenlung** [comment]
I have scores with multiple keyboards, and I think those tend to have multiple channels, so I don't believe guitars should be any exception, therefore I prefer channel change.

However, yes, Mixer certainly needs improvement in terms of layout. There were a few proposals.

**Comment by Jojo-Schmitz** [comment]
Keyboards have only one channel. Synthesizers have multiple, many. 
Acoustic guitar would get one (nylon, mute) or two (stell, mute and jazz) channels, only e-guits would getb

**Comment by Jojo-Schmitz** [comment]
Keyboards have only one channel. Synthesizers have multiple, many. 
Acoustic guitar would get one (nylon, mute) or two (stell, mute and jazz) channels, only e-guits would getb

**Comment by Jojo-Schmitz** [comment]
Keyboards have only one channel. Synthesizers have multiple, many. 
Acoustic guitar would get one (nylon, mute) or two (stell, mute and jazz) channels, only e-guits would get more, adding distortion,  overdrive and maybe harmonics

**Comment by Jojo-Schmitz** [comment]
Reg. cluttered mixer, on IRC @lasconic brought up the interesting idea to simply hide (or rather not show) those channels that are not used in the score, Score::updateChannel() does collect the info, and as far as I can see stores it in part.channelList.

**Comment by IsaacWeiss** [comment]
That's brilliant! Like all great ideas, in hindsight it seems so obvious that it's surprising it hasn't been that way all along.

**Comment by MarcSabatella** [comment]
Agreed on @lasconic idea.  If people were uncomfortable with losing the ability to pre-set the volume for channels they are using *yet*, we could always just have a checkbox "Display unused channels".

Instrument changes are nice, but they don't solve the clutter problem, they make it worse.  Because now every time you change between open and muted trumpet, or between one guitar sound and another, you add another channel.  Only if we added the ability to re-use channels would this be a viable solution.  My sense is that this wouldn't actually be that hard, but that it's not something to consider for 2.2 as it would probably create compatibility issues.

**Comment by Jojo-Schmitz** [comment]
Possible downside of naming the default channels: they now show with `partname()-channel()` (e.g. `violin-arco` or `trumpet-open`) in mixer.

**Comment by Jojo-Schmitz** [comment]
Travis mtest failures expected, need to be dealt with eventually, by adjusting the tests

**Comment by IsaacWeiss** [comment]
This was just requested with a precise description at https://musescore.org/en/node/210546#comment-825518.

**Comment by Jojo-Schmitz** [comment]
Rebases to build with recent change of `SubStyle` to `SubStyleId`

File: mscore/menus.cpp [hunk 1]
```diff
@@ -1435,6 +1435,33 @@ Palette* MuseScore::newTextPalette()
       stxt->setXmlText(tr("System Text"));
       sp->append(stxt, tr("System text"));
 
+      if (t != PaletteType::BASIC) {
+            StaffText* pz = new StaffText(SubStyleId::EXPRESSION, gscore);
```
Comment (Jojo-Schmitz) [inline mscore/menus.cpp]: `SubStyleId` would need to get changed to `SubStyle` when cherry-picking for a 2.x branch

File: mscore/editstyle.ui [hunk 2]
```diff
@@ -7661,7 +7661,7 @@
         </item>
        </layout>
       </widget>
-      <widget class="QWidget" name="page">
+      <widget class="QWidget" name="PageStaffText">
```
Comment (Jojo-Schmitz) [inline mscore/editstyle.ui]: This change would not cherry-pick to a 2.x branch, nor is needed there. Actually it is even really related to the topic of this PR...

**Comment by Jojo-Schmitz** [comment]
How to get the list in Synthesizer to update with the use of those channels, without having to close and reopen score?

**Comment by lgmventura** [comment]
About showing only the used channels in the mixer, it should be optional. I often configure my score on the mixer before I create anything and in this case I want to see all the tracks there. ;-)

**Comment by Jojo-Schmitz** [comment]
note to myself: rebase needed

**Comment by anatoly-os** [comment]
@Jojo-Schmitz only when #3898 is merged ;)

**Comment by Jojo-Schmitz** [comment]
Ah, yes

**Comment by Jojo-Schmitz** [comment]
The need for this came up again today, twice, in https://musescore.org/en/node/278042 and in https://musescore.org/en/node/99071#comment-868002

**Comment by Jojo-Schmitz** [comment]
_showing only the used channels in the mixer_ might have become obsolete with the new Mixer, which doesn't show channels by default at all. And with that the _How to get the list in Synthesizer to update with the use of those channels_ too

**Comment by Jojo-Schmitz** [comment]
Rebased.
Todo: add (more) channels for guitars, like 'muted', 'jazz', 'distortion', 'overdriven' and maybe 'harmonics'

**Comment by Jojo-Schmitz** [comment]
Rebased and added (more) channels for guitars, like 'muted', 'jazz', 'distortion', 'overdriven' and even 'harmonics'

**Pull request merged**